userspace/libsinsp/sinsp.cpp

```diff
@@ -1108,7 +1108,7 @@ void sinsp::get_procs_cpu_from_driver(uint64_t ts)
 	{
 		m_meinfo.m_cur_procinfo_evt = -1;
 
-		m_meinfo.m_piscapevt->ts = m_next_flush_time_ns - (ONE_SECOND_IN_NS + 1);
+		m_meinfo.m_piscapevt->ts = ts;
```

Comment:
Mmh is this right? I mean, i think the idea behind the old code was that:
`m_next_flush_time_ns` is exactly one second from now, so `m_next_flush_time_ns - (ONE_SECOND_IN_NS + 1);` is exactly 1ns before now.
I guess we might want it to be `m_next_flush_time_ns - ONE_SECOND_IN_NS + 1;` .
I am not sure, perhaps `ts` is already fine though.

Perhaps it would be even better to avoid setting event ts at this stage, and set it later as soon as the metaevt is retrieved from the `next` loop?

Reply:
Problem is that the result of `m_next_flush_time_ns - (ONE_SECOND_IN_NS + 1)` is in general less that `ts` by definitition. Considering that `ts` is the timestamp of the last received event, this always means creating an event "generated in the past".

I agree with you, a more generic fix would be to set the timestemp right within the `next()` method every time an internal/metaevent is fetched, but that would be a bigger change that may cause trouble  in the rest of libsinsp potentially. For now, i'd prefer to just set the timestamp here to the last event time, which is what we do for all metaevents except this one.

Reply:
Makes sense, thank you!